internal/execution/executor.go

```diff
@@ -58,31 +58,40 @@ func (e *Executor) Synthesize(ctx context.Context, env *Env) error {
 		return fmt.Errorf("building synthesizer input: %w", err)
 	}
 
-	output, err := e.Handler(ctx, syn, input)
-	if err != nil {
-		return fmt.Errorf("executing synthesizer: %w", err)
-	}
-	resultErr := findResultError(output)
+	var sliceRefs []*apiv1.ResourceSliceRef
+	output, handlerErr := e.Handler(ctx, syn, input)
 
-	err = e.preflightValidateResources(output)
-	if err != nil {
-		return err
-	}
+	// The handler error is either the error result emitted by the synthesizer
+	// or the handler's underlying "transport" error e.g. exec, json, etc.
+	if handlerErr != nil {
+		logger.Error(handlerErr, "unable to execute synthesizer")
 
-	var sliceRefs []*apiv1.ResourceSliceRef
-	if resultErr == nil {
-		sliceRefs, err = e.writeSlices(ctx, comp, output)
+		output = &krmv1.ResourceList{Results: []*krmv1.Result{{
+			Message:  "Synthesizer error: " + handlerErr.Error(),
```

Comment:
should we truncate here or rely on handler? Can we blow up a custom resoruce with a huge error?

Reply:
IMO it's fine to trust handler here since it's an implementation detail of the same package, just abstracted out for testing. Worst case the synthesizer update fails because it's too large which is fine